.github/workflows/update.yml

```diff
@@ -0,0 +1,52 @@
+name: Update flake inputs
+on:
+  schedule:
+    # Update every Sunday and Wednesday
+    - cron: "51 3 * * 0,3"
+  workflow_dispatch:
+
+jobs:
+  update:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+      - uses: cachix/install-nix-action@v13
+        with:
+          install_url: https://github.com/numtide/nix-unstable-installer/releases/download/nix-2.4pre20210415_76980a1/install
+          extra_nix_config: |
+            experimental-features = nix-command flakes
+      - name: Prepare flakebot branch
+        run: |
+          git config --local user.email "flakebot@users.noreply.github.com"
+          git config --local user.name "flakebot"
+          git checkout -B flake-update
+      - name: Update & commit lockfile
+        run: |
+          nix flake update --commit-lock-file
+          echo "commit_info<<EOF" >> $GITHUB_ENV
+          git log --format=%B -1 >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+      - name: Push changes
+        uses: ad-m/github-push-action@master
+        with:
+          # Use a custom token so we can trigger other workflows upon push to master
+          github_token: ${{ secrets.FLAKEBOT_TOKEN }}
+          branch: flake-update
+          force: true
+      - name: pull-request-action
+        uses: vsoch/pull-request-action@1.0.15
+        env:
+          GITHUB_TOKEN: ${{ secrets.FLAKEBOT_TOKEN }}
+          PULL_REQUEST_FROM_BRANCH: flake-update
+          PULL_REQUEST_BRANCH: master
+          PULL_REQUEST_UPDATE: true
+          PASS_ON_ERROR: true
```

Comment:
Why do we need this?

Reply:
This will open a new pull request against the `master` branch, or, if one exists already, update the existing one. `PASS_ON_ERROR` is set to true since this action will always be run, regardless of whether the flake was updated during the workflow. So if there were no changes, this step would fail, but we don't want the workflow to fail because of that.

Reply:
Can we instead check if there was any update, and optionally skip this step? Because I'm not sure if we should have a blanked ignore of all errors. Could be hard to notice when there are actual issues.

Reply:
Hmm, this will be a lot harder than just checking `commit_info`, since that is literally the commit info of HEAD, regardless of whether a flake update was committed or not. I suppose I can check if a commit was made by comparing against [`github.sha`](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context), though.

Reply:
After thinking about it some more, I think a better way to do this would be to ask the author of the action to let us specify to pass just if there are no new commits, that way I don't have to "reinvent the wheel". I can open an issue for that.

Reply:
Action author has suggested that I just implement this logic in this workflow's step, so I have pushed 0be2c37fd5beeaf60366679316f350ab4f5fef61. @asymmetric can you check if this looks right?